syne_tune/optimizer/schedulers/multiobjective/linear_scalarizer.py

```diff
@@ -0,0 +1,160 @@
+    https://stackoverflow.com/questions/3844801/check-if-all-elements-in-a-list-are-identical
+    """
+    g = groupby(iterable)
+    return next(g, True) and not next(g, False)
+
+
+class LinearScalarizedScheduler(TrialScheduler):
+    """Scheduler with a Linear Scalarization of multiple objectives.
+    This method optimizes a single objective equal to the linear scalarization of given two objectives.
+    The scalarized single objective is named: 'scalarized_<metric1>_<metric2>_..._<metricN>'
+
+
+    :param base_scheduler: Class for the single-objective scheduler to be used on the scalarized objective.
+        It will be initialized inside this Scheduler
+    :param config_space: Configuration space for evaluation function
+    :param metric: Names of metrics to optimize
+    :param mode: Modes of metrics to optimize (min or max). All must be matching.
+    :param scalarization_weights: Weights used to scalarize objectives, if None an array of 1s is used
+    :param kwargs: Additional arguments to base_scheduler
+    """
+
+    scalarization_weights: np.ndarray
+    single_objective_metric: str
+    base_scheduler: TrialScheduler
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        mode: Union[List[str], str] = "min",
+        scalarization_weights: Union[np.ndarray, List[float]] = None,
+        base_scheduler: Callable[[Any], TrialScheduler] = FIFOScheduler,
+        **kwargs,
+    ):
+        super(LinearScalarizedScheduler, self).__init__(config_space)
+        if scalarization_weights is None:
+            scalarization_weights = np.ones(shape=len(metric))
+        self.scalarization_weights = np.asarray(scalarization_weights)
+
+        self.metric = metric
+        self.mode = mode
+
+        self.single_objective_metric = f"scalarized_{'_'.join(metric)}"
+        assert len(mode) >= 1, "At least one mode must be provided"
+        if isinstance(mode, Iterable):
+            assert _all_equal(
+                mode
+            ), "Modes must be the same, use positive/negative scalarization_weights to change relative signs"
+            single_objective_mode = next(x for x in mode)
+
+        self.base_scheduler = base_scheduler(
+            config_space=config_space,
+            metric=self.single_objective_metric,
+            mode=single_objective_mode,
+            **kwargs,
+        )
+
+    def _scalarized_metric(self, result: Dict[str, Any]) -> float:
+        if isinstance(self.base_scheduler, FIFOScheduler):
+            FIFOScheduler._check_keys_of_result(result, self.metric_names())
+
+        mo_results = np.array([result[item] for item in self.metric_names()])
+        logger.debug(
+            f"Scalarizing {self.metric_names()} into {self.single_objective_metric} using linear combination"
+        )
+        return np.sum(np.multiply(mo_results, self.scalarization_weights))
+
+    def suggest(self, trial_id: int) -> Optional[TrialSuggestion]:
```

Comment:
Maybe better to overwrite `_suggest`

Reply:
Why do you think so? Both should result in the same behaviour if `base_scheduler` only overrides `_suggest` but in case `base_scheduler` overrides `sugges` we want to call it (`suggest`) to ensure desired behaviour.